src/Transactions/UpdateContract.cs

```diff
@@ -0,0 +1,124 @@
+    /// .</param>
+    public PreparedAccountTransaction Prepare(
+        AccountAddress sender,
+        AccountSequenceNumber sequenceNumber,
+        Expiry expiry,
+        EnergyAmount energy
+    ) => new(sender, sequenceNumber, expiry, new(_baseCost.Value + energy.Value), this);
+
+    /// <summary>
+    /// The transaction specific cost for submitting this type of
+    /// transaction to the chain.
+    ///
+    /// This should reflect the transaction-specific costs defined here:
+    /// https://github.com/Concordium/concordium-base/blob/78f557b8b8c94773a25e4f86a1a92bc323ea2e3d/haskell-src/Concordium/Cost.hs
+    /// </summary>
+    private static readonly EnergyAmount _baseCost = new(300);
+
+    /// <summary>The account transaction type to be used in the serialized payload.</summary>
+    private const byte TransactionType = (byte)Types.TransactionType.Update;
+
+    /// <summary>
+    /// Gets the size (number of bytes) of the payload.
+    /// </summary>
+    internal override PayloadSize Size() => new(sizeof(TransactionType) + CcdAmount.BytesLength + ContractAddress.BytesLength + this.ReceiveName.SerializedLength() + this.Parameter.SerializedLength());
+
+    /// <summary>
+    /// Create a "update contract" payload from a serialized as bytes.
+    /// </summary>
+    /// <param name="bytes">The "update contract" payload as bytes.</param>
+    /// <param name="output">Where to write the result of the operation.</param>
+    public static bool TryDeserial(ReadOnlySpan<byte> bytes, out (UpdateContract? payload, string? Error) output)
+    {
+        var minLength = sizeof(TransactionType) + CcdAmount.BytesLength + ContractAddress.BytesLength + ReceiveName.MinSerializedLength + Parameter.MinSerializedLength;
+        if (bytes.Length < minLength)
+        {
+            var msg = $"Invalid input length in `UpdateContract.TryDeserial`. Expected at least {minLength}, found {bytes.Length}";
+            output = (null, msg);
+            return false;
+        }
+
+        if (bytes[0] != TransactionType)
+        {
+            var msg = $"Invalid transaction type in `UpdateContract.TryDeserial`. Expected {TransactionType}, found {bytes[0]}";
+            output = (null, msg);
+            return false;
+        }
+
+        var remaining_bytes = bytes[sizeof(TransactionType)..];
+
+        if (!CcdAmount.TryDeserial(remaining_bytes, out var amount))
+        {
+            output = (null, amount.Error);
+            return false;
+        };
+        remaining_bytes = remaining_bytes[(int)CcdAmount.BytesLength..];
+
+        if (!ContractAddress.TryDeserial(remaining_bytes, out var address))
+        {
+            output = (null, address.Error);
+            return false;
+        };
+        remaining_bytes = remaining_bytes[(int)ContractAddress.BytesLength..];
+
+        if (!ReceiveName.TryDeserial(remaining_bytes, out var receiveName) || receiveName.receiveName == null)
```

Comment:
I need to know that it is not null already below this if statement, because I use it to create the span of remaining bytes.
It does stick out and looks annoying, though.

Reply:
Can’t you make it part of `ReceiveName.TryDeserial` to ensure it isn’t null and then use `!`. Seems wrong to me that you have the function but still needs to validate the output.

Reply:
I have removed the check here